### ISSUE TITLE
SiS generate new StateCode for upleveled logins

### ISSUE THREAD

**Issue by bramleyjl** [department-of-veterans-affairs/va.gov-team#47992]
Due to SiS validating & deleting the vets-api-generated `StateCode` during the `/callback` request, a new `StateCode` for upleveled logins (2 trips) is needed to prevent validation errors when checking the now-deleted code during the second request.

**Comment by joeniquette** [comment]
Regarding the production test conducted on 10/7:
- This ticket is a fix to the bug found during the production test
- We ran the prod test on Friday 10/7 to try and see if we could fix a bug regarding multiple uses of the same state value during the code change step with IDme and login.gov. We called these errors "token exchange error" and "userinfo request error". The issues were around users appearing to exchange state values and auth codes multiple times. Our first fix attempted to only allow once use of the state code. During upleveling we use the state code a second time. So we made the change is to re-generate the state code during the upleveling step.
  - This change prevented 40 production users from authenticating on VA.gov, which lasted about 15 minutes. The error was on upleveling users to loa3/ial2
